tests/unit/machine/EnsembleMachine_unittest.cc

```diff
@@ -0,0 +1,59 @@
+
+#include <gtest/gtest.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/machine/EnsembleMachine.h>
+#include <shogun/ensemble/MajorityVote.h>
+#include <shogun/evaluation/MulticlassAccuracy.h>
+#include <shogun/features/DataGenerator.h>
+#include <shogun/mathematics/NormalDistribution.h>
+#include <shogun/multiclass/MulticlassLibLinear.h>
+#include <shogun/multiclass/MulticlassOCAS.h>
+using namespace shogun;
+
+TEST(EnsembleMachine, train)
+{
+	int32_t seed = 100;
+	index_t num_vec = 10;
+	index_t num_feat = 3;
+	index_t num_class = num_feat; // to make data easy
+	float64_t distance = 15;
+
+	// create some linearly seperable data
+	SGMatrix<float64_t> matrix(num_class, num_vec);
+	SGMatrix<float64_t> matrix_test(num_class, num_vec);
+	auto labels = std::make_shared<MulticlassLabels>(num_vec);
+	auto labels_test = std::make_shared<MulticlassLabels>(num_vec);
+	std::mt19937_64 prng(seed);
+	NormalDistribution<float64_t> normal_dist;
+	for (index_t i = 0; i < num_vec; ++i)
+	{
+		index_t label = i % num_class;
+		for (index_t j = 0; j < num_feat; ++j)
+		{
+			matrix(j, i) = normal_dist(prng);
+			matrix_test(j, i) = normal_dist(prng);
+			labels->set_label(i, label);
+			labels_test->set_label(i, label);
+		}
+
+		/* make sure data is linearly seperable per class */
+		matrix(label, i) += distance;
+		matrix_test(label, i) += distance;
+	}
```

Comment:
have you tried to look at the data that is being generated here? I am not sure I understand the logic..

Reply:
I copy those code from `MulticlassLibLinear_unittest`.